### PR TITLE
Refactor blueprint route validation and generate events on errors

### PR DESCRIPTION
**Pull request by ramr**
As per #20646 follow up to refactor the validation and generate events on errors.

/cc @openshift/sig-network-edge 

*Edited* Update PR to remove event generation code as per the standup discussion.

### CODE REVIEW

File: pkg/router/template/configmanager/haproxy/blueprint_plugin.go [hunk 1]
```diff
@@ -53,3 +62,29 @@ func (p *BlueprintPlugin) Commit() error {
...
+			Component: fmt.Sprintf("blueprint route %s", name),
+		},
+		Type: v1.EventTypeWarning,
+
+		ReportingController: "cm.haproxy.router.openshift.io/blueprints",
+	}
+
+	if _, err := p.client.Events(route.Namespace).Create(e); err != nil {
+		glog.Errorf("adding blueprint route error event: %v", err)
+		return
```
Comment (ironcladlou) [inline pkg/router/template/configmanager/haproxy/blueprint_plugin.go]: Superfluous `return`

**Comment by ramr** [comment]
Closing this out as the router code moved to another repo and a new PR was opened for this.
Superseded by https://github.com/openshift/router/pull/10

**Pull request closed without merging**